components/user_settings/user_settings_security/user_settings_security.jsx

```diff
@@ -1099,6 +1109,28 @@ export default class SecurityTab extends React.Component {
         this.handleCancelConfirm();
     }
 
+    activateToken = async (tokenId) => {
+        const {error} = await this.props.actions.enableUserAccessToken(tokenId);
+        if (error) {
+            this.setState({serverError: error.message});
+        } else {
+            const userId = this.props.user ? this.props.user.id : '';
+            this.props.actions.getUserAccessTokensForUser(userId, 0, 200);
```

Comment:
Is getting them all again necessary? I thought the action would propagate any needed changes to the store

Reply:
I had a lot of trouble getting the component to update on enable/disable until I added this. Maybe because the change was too deep? I'm open to other suggestions though.

Reply:
Hmmm ok, I think I'll pull your changes locally and take a quick look later today. Might just be missing something subtle somewhere

Reply:
Fix here mattermost/mattermost-redux#304, you were missing the action handling in some reducers

Reply:
Ah yep, I see. I'll revise here to adjust.